pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -44,12 +44,12 @@ let
   version = {
     # day of the snapshot being taken
     year = "2024";
```

Comment:
Shouldn't this be changed to 2025? (not that it really matters)

Reply:
This is not texlive 2025, this is texlive released in 2024 with all updates to packages accumulated.

Reply:
The texliveYear is defined below this, as a separate variable.

Reply:
> Shouldn't this be changed to 2025? (not that it really matters)

Yes, I have changed day and month, so for consistently I should have typed 2025 as well. It does, indeed, not matter at all.

Reply:
> The texliveYear is defined below this, as a separate variable.

We should fix it then, it shouldn't show up anywhere for "final" mode.

Reply:
Not sure how that could be done? `texliveYear` is required for the first mirror URL below, and `year` for the last mirror URL. In fact that means the year really needs to be updated to 2025 (unless the last mirror URL doesn't work for the final release?)

Reply:
When `final = true`, the year/month/day information is completely discarded (if not, that's a bug!).

Reply:
Well then there is a bug. See the mirror list definition directly below, where year/month/day are used.

Reply:
> When `final = true`, the year/month/day information is completely discarded (if not, that's a bug!).

Now, actually it doesn't! That `lib.optionals` should really be `if ... then ... else ...`!
https://github.com/NixOS/nixpkgs/blob/acc137d81e3a6d099af71eea0c8fbb436fb36d6c/pkgs/tools/typesetting/tex/texlive/default.nix#L60